practice and Hands on lab at the same time:

In [1]:
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv("k-means.csv")

In [22]:
mean = np.mean(df['income'])
median = np.median(df['income'])
mode = np.median(df['income'])

max = np.max(df['income'])
min = np.min(df['income'])

print(f"mean: {mean:.2f} ")
print("median: ", median)
print("mode: ", mode)

print ("max: ",max)
print("min: ",min)

mean: 90431.82 
median:  67500.0
mode:  67500.0
max:  162000
min:  45000


In [24]:
std = np.std(df['income'])
var = np.var(df['income'])
q1 , q3 = np.percentile(df['income'], [25, 75])
IQR = q3-q1

print(f"standered deviation: {std:.2f}")
print(f"variance: {var:.2f}")
print(f"1st quartile: {q1}, 3rd quartile: {q3}")
print("IQR: ", IQR)

standered deviation: 42505.69
variance: 1806733987.60
1st quartile: 58500.0, 3rd quartile: 135250.0
IQR:  76750.0


using the mean only makes sense when data isn't skewed  or is containing outliers, that is when it can be accurate.

while on the other side median is more suitable in such scenarios since it takes the middle value. rendering it more robust and closer to the typical value in the data.

The column centre in this case is most accuratly represented by median or mode at around 67500.
 it is also showing some skewness to the right via the mean gravitating more towards a higher than median value 90432.

As for the Overall range of the data spread it seems to vary noticably strating from a min of 45000 up to a max of 162000.

 in addition the middle(50%) range which is represented via IQR has a value of 76750.
 
 and it also has a standered deviation value of 42505, which indicates that value point away from mean. 
 
 All of these indicate a wide spread data. 